src/page/ConfigPageLayout/tabs/SecretsManagementTab/SecretCard.test.tsx

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import { act, fireEvent, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { SecretWithMetadata } from './types';
+import { formatDate } from 'utils';
+
+import { SecretCard } from './SecretCard';
+
+// Mock the formatDate utility
+jest.mock('utils', () => ({
+  formatDate: jest.fn((date) => date.toString()),
+}));
+
+const writeText = jest.fn();
+
+Object.assign(navigator, {
+  clipboard: {
+    writeText,
+  },
+});
+
+Object.defineProperty(window, 'isSecureContext', {
+  value: true,
+});
+
+describe('SecretCard', () => {
+  const mockSecret: SecretWithMetadata = {
+    uuid: 'secret-123',
+    name: 'Test Secret',
+    description: 'Test Description',
+    created_at: Number(new Date('2024-01-01')),
+    created_by: 'test-user',
+    labels: [
+      { name: 'env', value: 'prod' },
+      { name: 'type', value: 'api-key' },
+    ],
+    modified_at: Number(new Date('2024-01-02')),
+    org_id: 1,
+    stack_id: 1,
+  };
+
+  const defaultProps = {
+    secret: mockSecret,
+    onEdit: jest.fn(),
+    onDelete: jest.fn(),
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should render secret details correctly', () => {
+    render(<SecretCard {...defaultProps} />);
+
+    // Check basic information
+    expect(screen.getByText('Test Secret')).toBeInTheDocument();
+    expect(screen.getByText('Test Description')).toBeInTheDocument();
+    expect(screen.getByText('secret-123')).toBeInTheDocument();
+    expect(screen.getByText(/test-user/)).toBeInTheDocument();
+
+    // Check labels
+    expect(screen.getByText('env: prod')).toBeInTheDocument();
+    expect(screen.getByText('type: api-key')).toBeInTheDocument();
+  });
+
+  it('should call onEdit when edit button is clicked', () => {
+    render(<SecretCard {...defaultProps} />);
+
+    const editButton = screen.getByRole('button', { name: 'Edit Test Secret' });
+    fireEvent.click(editButton);
+
+    expect(defaultProps.onEdit).toHaveBeenCalledWith('secret-123');
+  });
+
+  it('should call onDelete when delete button is clicked', () => {
+    render(<SecretCard {...defaultProps} />);
+
+    const deleteButton = screen.getByRole('button', { name: 'Delete Test Secret' });
+    fireEvent.click(deleteButton);
```

Comment:
For component interactions, we [should be using](https://ph-fritsche.github.io/blog/post/why-userevent) `userEvent` instead of `fireEvent`